### PR TITLE
Some fixes in dump of annotations

### PR DESCRIPTION
**Pull request by bsekachev**

### CODE REVIEW

File: cvat/apps/engine/annotation.py [hunk 1]
```diff
@@ -818,6 +818,9 @@ def merge(self, objects, start_frame, overlap):
 
         # Nothing to merge here. Just add all int_objects if any.
         if not old_objects_by_frame or not int_objects_by_frame:
+            for old_obj in self.objects:
```
Comment (nmanovic) [inline cvat/apps/engine/annotation.py]: need to use old_objects_by_frame.values()

File: cvat/apps/engine/annotation.py [hunk 2]
```diff
@@ -1002,7 +1005,11 @@ def _calc_objects_similarity(obj0, obj1, start_frame, overlap):
                     if shape0["outside"] != shape1["outside"]:
                         error += 1
                     else:
-                        error += 1 - ShapeManager._calc_objects_similarity(shape0, shape1)
+                        shape0["label_id"] = obj0["label_id"]
+                        shape1["label_id"] = obj1["label_id"]
+                        error += 1 - ShapeManager._calc_objects_similarity(shape0, shape1, start_frame, overlap)
```
Comment (nmanovic) [inline cvat/apps/engine/annotation.py]: Let's modify _calc_objects_similarity (get labeled_id -1 by default)

File: cvat/apps/engine/annotation.py (hunk 1 above)
Comment (bsekachev) [reply cvat/apps/engine/annotation.py]: Done.

File: cvat/apps/engine/annotation.py (hunk 2 above)
Comment (bsekachev) [reply cvat/apps/engine/annotation.py]: Done

**Pull request merged**